### PR TITLE
【Error Message No. 4】 paddle/fluid/pir/transforms/* fix errors

### PR DESCRIPTION
**Pull request by enkilee**
### PR types

Others
### PR changes

Others
### Description

* https://github.com/PaddlePaddle/Paddle/pull/62814
fix 3 errors

### CODE REVIEW

File: paddle/fluid/pir/transforms/shape_optimization_pass.cc [hunk 1]
```diff
@@ -245,7 +245,7 @@ class ShapeOptimizationPass : public pir::Pass {
         << "===================== ShapeOptimizationPass Run start... "
            "=====================";
     auto module_op = op->dyn_cast<pir::ModuleOp>();
-    PADDLE_ENFORCE_EQ(module_op.name(),
+    PADDLE_ENFORCE_NE(module_op.name(),
                       "builtin.module",
                       phi::errors::InvalidArgument(
                           "ShapeOptimizationPass should run on module op."));
```
Comment (yuanlehome) [inline paddle/fluid/pir/transforms/shape_optimization_pass.cc]: 正确的写法是 
PADDLE_ENFORCE_NOT_NULL(module_op,
                      phi::errors::InvalidArgument(
                          "ShapeOptimizationPass should run on module op."));

File: paddle/fluid/pir/transforms/sub_graph_extract_pass.cc [hunk 2]
```diff
@@ -46,7 +46,7 @@ class SubGraphExtractPass : public pir::Pass {
 
   void Run(pir::Operation* op) override {
     auto module_op = op->dyn_cast<pir::ModuleOp>();
-    PADDLE_ENFORCE_EQ(module_op.name(),
+    PADDLE_ENFORCE_NE(module_op.name(),
```
Comment (yuanlehome) [inline paddle/fluid/pir/transforms/sub_graph_extract_pass.cc]: 同上

File: paddle/fluid/pir/transforms/fusion/conv2d_add_act_fuse_pass.cc [hunk 3]
```diff
@@ -44,8 +44,8 @@ class Conv2dAddActFusePattern
 
     pir::Value add_input = op.x();
     PADDLE_ENFORCE_EQ(
-        add_input && conv2d_out,
-        true,
+        add_input.type(),
+        conv2d_out.type(),
         phi::errors::PreconditionNotMet("The type of add input should be the "
                                         "same as the type of conv2d's out."));
```
Comment (yuanlehome) [inline paddle/fluid/pir/transforms/fusion/conv2d_add_act_fuse_pass.cc]: 正确的写法是
PADDLE_ENFORCE_EQ(
        add_input,
        conv2d_out,
        phi::errors::PreconditionNotMet("The type of add input should be the "
                                        "same as the type of conv2d's out."));

Comment (enkilee) [reply paddle/fluid/pir/transforms/fusion/conv2d_add_act_fuse_pass.cc]: 收到

File: paddle/fluid/pir/transforms/sub_graph_extract_pass.cc (hunk 2 above)
Comment (enkilee) [reply paddle/fluid/pir/transforms/sub_graph_extract_pass.cc]: 收到

File: paddle/fluid/pir/transforms/shape_optimization_pass.cc (hunk 1 above)
Comment (enkilee) [reply paddle/fluid/pir/transforms/shape_optimization_pass.cc]: 收到

**Comment by luotao1** [comment]
@enkilee 需要解决下冲突

**Review by yuanlehome — APPROVED** [review]

**Review by Galaxy1458 — APPROVED** [review]

**Pull request merged**